server/handlers/result_handlers.py

```diff
@@ -75,7 +75,7 @@ class LSEBatchItem(BaseModel):
     )
 
     task_id: int
-    output: str
+    output: Optional[str]
```

Comment:
Notably, this is `None` when there's an error, and it ends up surfacing as a pretty cryptic pickling error (at least at the final stacktrace, there are logs beforehand that show the problem)
Making this Optional makes this a bit easier to trace (and we'll need it anyway once we improve our error handling)